### ISSUE TITLE
SuspendFunSwallowedCancellation does not work when Throwable is caught

### ISSUE THREAD

**Issue by matejdro** [detekt/detekt#9185]
## Expected Behavior
Following code should trip `SuspendFunSwallowedCancellation`:

```kotlin
suspend fun test() {
   try {
      awaitCancellation()
   } catch (e: Throwable) {

   }
}
```

## Observed Behavior

Rule is not tripped

## Context

Catching `Exception` instead of `Throwable` does work, so it seems rule does not like catching Throwables?

## Your Environment

* Version of detekt used: `2.0.0-alpha.2`
